[APM] Service maps - Fix missing ML status for services with jobs but no anomalies

ogupte opened the pull request:
Closes #68485 by: 
 - use the ML plugin to query for all APM jobs
  - inspect the ml job groups to find all jobs related to a particular service
  - use the mlAnomalySearch client to get ml job buckets with the max anomaly score &  actual/median values for the ML description
  - return the relevant ML job with the max anomaly score for a service
  - indicate to the user that no anomalies were found for a service with an ml job
![Screen Shot 2020-06-09 at 9 12 49 AM](https://user-images.githubusercontent.com/1967266/84173128-b11feb80-aa31-11ea-9618-cb563f2d9ee6.png)

ogupte committed:
Closes #68485 by:
  - use the ML plugin to query for all APM jobs
  - inspect the ml job groups to find all jobs related to a particular service
  - use the mlAnomalySearch client to get ml job buckets with the max anomaly score
  - query for the model_plot buckets to obtain actual/median values for the ML description
  - return the relevant ML job with the max anomaly score for a service
  - indicate to the user that no anomalies were found for a service with an ml job

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/)
* Commit: acc3db620f9cb24e0db59d67c21e85bc9b4ca87a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/","number":"52304","commit":"acc3db620f9cb24e0db59d67c21e85bc9b4ca87a"}]}
PIPELINE-->

jgowdyelastic commented on x-pack/plugins/apm/server/lib/helpers/setup_request.ts:
```diff
@@ -104,3 +107,25 @@ export async function setupRequest<TParams extends SetupRequestParams>(
+    mlJobs: async (
+      jobId?: string | string[]
+    ): Promise<AnomalyDetectionJob[]> => {
+      const mlJobsResponse = await mlClient('ml.jobs', { jobId });
```
To comply with our current spaces work, external plugins should no long be calling the ML elasticsearch endpoints directly. 
They need to instead be calling the functions shared in out setup contract.
This call is covered by `jobs` inside `anomalyDetectorsProvider`.

Note, I just noticed that the return type for `jobs` is using an `any` rather than the correct return return type of  `{ count: number; jobs: Job[] }`
This will be corrected very soon as part of our on going spaces work.

ogupte replied:
in the latest commit i updated the return type of that API within ML

jgowdyelastic commented on x-pack/plugins/ml/server/shared.ts:
```diff
@@ -5,3 +5,4 @@
  */
 
 export * from '../common/types/anomalies';
+export { Job } from '../common/types/anomaly_detection_jobs';
```
This could also be `*` as we might has well share all of our anomaly detection job types for future use.

formgeist commented on x-pack/plugins/apm/public/components/app/ServiceMap/Popover/Contents.tsx:
```diff
@@ -116,6 +116,14 @@ const ANOMALY_DETECTION_DISABLED_TEXT = i18n.translate(
+  'xpack.apm.serviceMap.anomalyDetectionPopoverNoData',
+  {
+    defaultMessage:
+      'No anomaly score found with the current filters. See details in the anomaly explorer:',
```
```suggestion
      'We couldn't find any anomaly score within the current time range. See details in the anomaly explorer.',
```

@ogupte do you think this copy covers the scenario? I'm just trying to make a little more assuring while still being descriptive about they might be looking at the wrong time range or there's something wrong with the data feed. Also since we're currently not taking environment filter into account, we best just clarify that it's simply the time range that we're looking at.

bmorelli25 replied:
Is it only possible to have one anomaly score per service in a given time frame? If so, what about `an` instead of `any`?

```
We couldn't find an anomaly score within the current time range. See details in the anomaly explorer.
```

ogupte replied:
We display the max anomaly score within the time range here, so just 1 score. If any service has more than 1 ml jobs associated with it, it will only display the one with the higher score.

bmorelli25 replied:
Neat. I think this works then: "We couldn't find an anomaly score within the selected time range. See details in the anomaly explorer."

formgeist reviewed (commented):
Just a suggestion on the empty state copy

ogupte committed:
- Use the anomalyDetectorsProvider jobs API rather than the search endpoint directly
- Defines a specific return type for the ml jobs api
- Update the empty anomaly data message

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/)
* Commit: 08b94cfe9f947f40dbdf43afe053b47d630153ea

### History
* :green_heart: [Build #52304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/) succeeded acc3db620f9cb24e0db59d67c21e85bc9b4ca87a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/","number":"52304","commit":"acc3db620f9cb24e0db59d67c21e85bc9b4ca87a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/","number":"52453","commit":"08b94cfe9f947f40dbdf43afe053b47d630153ea"}]}
PIPELINE-->

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -0,0 +1,158 @@
+      (apmJobCategory) => apmJobCategory !== undefined
+    ) as ApmMlJobCategory[];
+  const apmJobIds = apmMlJobs.map((job) => job.job_id);
+  const rangeQuery = { range: rangeFilter(start, end, 'timestamp') };
```
I think this is still relevant:
https://github.com/elastic/kibana/pull/65217#discussion_r420484752

jgowdyelastic reviewed (approved):
ML changes LGTM

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -0,0 +1,158 @@
+const getApmMlJobCategory = (
+  mlJob: AnomalyDetectionJob,
+  serviceNames: string[]
+) => {
```
Weren't we aiming to use the parameterised jobs for 7.9 in which case this is no longer needed?

ogupte replied:
it's out of scope for the 7.8 fix, so if we go down that route for 7.9, we'll remove this

sqren replied:
Oh, sorry. Missed this was for 7.8!

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -0,0 +1,158 @@
+    ({ jobId, timestamp }) => {
+      return (async () => {
+        try {
+          const modelPlotResponse = await ml.mlSystem.mlAnomalySearch({
```
It looks like this is invoked inside a nested iteration. How many times is it called? Preferably we should be able to get anomalies with a single es request. is that no longer the case?

ogupte replied:
it fetches the matching `model_plot` bucket by timestamp. and it fetches each one in a separate request. There wasn't a way to do it all in one request unfortunately, but this search is super fast since there's no scoring, and only returns the 1 hit.

ogupte replied:
One alternative is to do the fetch when the user clicks on on the service to open the popover, and return it along with the other metrics.

ogupte replied:
After going down this path, i realized that we no longer needed to use the `bucket` & `model_plot` result types in the anomalies query, So i returned to using the previous `record` type, which includes the `actual` & `typical` values in the same document. This prevents the need to make any extra requests, so this issue is now resolved.

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -0,0 +1,158 @@
+    jobId: string;
+    anomalyScore: number;
+    timestamp: number;
+  }> = response.aggregations.jobs.buckets.map((jobBucket: any) => {
```
`any`

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -0,0 +1,158 @@
+  return array.filter(
+    (element) => element !== undefined || element !== null
+  ) as Array<NonNullable<T>>;
+}
```
Instead of overriding the compiler with `as` it's normally better to do something like:

```ts
function filterNonNullable<T>(value: T | null | undefined): value is T {
    return value !== null && value !== undefined;
}

// ...

return array.filter(filterNonNullable)
```

It uses the `is` syntax which is not bulletproof but better than `as` imo.

ogupte replied:
Thanks for the tip. What about in-lining it since it's only used in 1 place?
```
  const anomalyModelValues = (
    await Promise.all(anomalyModelValuePromises)
  ).filter(<T>(value: T | undefined): value is T => value !== undefined);
```

sqren replied:
I find it more readable in a separate function but no big deal.

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_map.ts:
```diff
@@ -129,58 +132,31 @@ async function getServicesData(options: IEnvOptions) {
-    getServicesData(options),
-    getAnomaliesData(options),
-  ]);
+    ServicesResponse
```
Is the type not inferred?

ogupte replied:
It might be able to be cleaned up now, but at first writing, the compiler would fail with a "excessively deep" TS error.

ogupte committed:
Code and types cleanup

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/)
* Commit: bd1ca55211d03b14d95d0de2a349c98acce54f87

### History
* :green_heart: [Build #52453](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/) succeeded 08b94cfe9f947f40dbdf43afe053b47d630153ea
* :green_heart: [Build #52304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/) succeeded acc3db620f9cb24e0db59d67c21e85bc9b4ca87a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/","number":"52304","commit":"acc3db620f9cb24e0db59d67c21e85bc9b4ca87a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/","number":"52453","commit":"08b94cfe9f947f40dbdf43afe053b47d630153ea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/","number":"52525","commit":"bd1ca55211d03b14d95d0de2a349c98acce54f87"}]}
PIPELINE-->

ogupte committed:
Return to using record result type on anomaly queries. These are the
same values used in the anomaly explorer and it includes actual &
typical values which greatly improve performance of the previous query.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/)
* Commit: 106d0f28fd7ea147c61605eac732bba2ede133e4

### History
* :green_heart: [Build #52525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/) succeeded bd1ca55211d03b14d95d0de2a349c98acce54f87
* :green_heart: [Build #52453](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/) succeeded 08b94cfe9f947f40dbdf43afe053b47d630153ea
* :green_heart: [Build #52304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/) succeeded acc3db620f9cb24e0db59d67c21e85bc9b4ca87a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/","number":"52304","commit":"acc3db620f9cb24e0db59d67c21e85bc9b4ca87a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/","number":"52453","commit":"08b94cfe9f947f40dbdf43afe053b47d630153ea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/","number":"52525","commit":"bd1ca55211d03b14d95d0de2a349c98acce54f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/","number":"52637","commit":"106d0f28fd7ea147c61605eac732bba2ede133e4"}]}
PIPELINE-->

ogupte committed:
- If anomaly data is missing show a gray border around node

smith commented on x-pack/plugins/apm/public/components/app/ServiceMap/Popover/Contents.tsx:
```diff
@@ -116,6 +116,20 @@ const ANOMALY_DETECTION_DISABLED_TEXT = i18n.translate(
+
+function getDisplayedAnomalyScore(score: number) {
+  if (score > 0 && score < 1) {
+    return '<1';
```
`<1` looks a little cramped. Can we put a space? `< 1`

smith reviewed (approved):
A few nits in the comments, but works well for me. Thanks!

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/)
* Commit: 143c0a8f550b776b0fc22e8f24009fcf4281c707
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/execution/node/527/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/execution/node/702/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/testReport)

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/endpoint/metadata·ts.apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging"</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_endpoint_metadata·ts/Kibana_Pipeline___kibana_xpack_agent___apis_Endpoint_plugin_test_metadata_api_POST__api_endpoint_metadata_when_index_is_not_empty__before_all__hook_for__metadata_api_should_return_one_entry_for_each_host_with_default_paging_)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:09:19]           └-: Endpoint plugin
[00:09:19]             └-> "before all" hook
[00:09:19]             └-> "before all" hook
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [events-mappings]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [logs-settings]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [metrics-settings]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [logs-mappings]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [events-settings]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [metrics-mappings]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [events]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics]
[00:09:19]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [logs]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-endpoint.metadata]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-endpoint.policy]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-endpoint.telemetry]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [events-endpoint]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.core]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.diskio]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.fsstat]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [logs-system.auth]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.cpu]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.filesystem]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.load]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.entropy]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.network]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.memory]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.network_summary]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.process_summary]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.raid]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.socket]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.process]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.service]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.socket_summary]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [logs-system.syslog]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.users]
[00:09:20]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.uptime]
[00:09:25]             └-: test metadata api
[00:09:25]               └-> "before all" hook
[00:09:25]               └-: POST /api/endpoint/metadata when index is not empty
[00:09:25]                 └-> "before all" hook
[00:09:25]                 └-> "before all" hook
[00:09:25]                   │ info [endpoint/metadata/api_feature] Loading "mappings.json"
[00:09:25]                   │ info [endpoint/metadata/api_feature] Loading "data.json"
[00:09:25]                   └- ✖ fail: "apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging""
[00:09:25]                   │

```

#### Stack Trace
```
{ Error: [illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition
    at respond (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:349:15)
    at checkRespForFailure (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:306:7)
    at HttpConnector.<anonymous> (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/connectors/http.js:173:7)
    at IncomingMessage.wrapper (/dev/shm/workspace/kibana/node_modules/elasticsearch/node_modules/lodash/lodash.js:4929:19)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  status: 400,
  displayName: 'BadRequest',
  message:
   '[illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition',
  path: '/metrics-endpoint.metadata-default-1',
  query: {},
  body:
   { error:
      { root_cause: [Array],
        type: 'illegal_argument_exception',
        reason:
         'mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition' },
     status: 400 },
  statusCode: 400,
  response:
   '{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"}],"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"},"status":400}',
  toString: [Function],
  toJSON: [Function] }
```
</details>

---

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/endpoint/metadata·ts.apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging"</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_endpoint_metadata·ts/Kibana_Pipeline___kibana_xpack_agent___apis_Endpoint_plugin_test_metadata_api_POST__api_endpoint_metadata_when_index_is_not_empty__before_all__hook_for__metadata_api_should_return_one_entry_for_each_host_with_default_paging__2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:10:03]           └-: Endpoint plugin
[00:10:03]             └-> "before all" hook
[00:10:03]             └-> "before all" hook
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [events-settings]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [logs-mappings]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [events-mappings]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [metrics-settings]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [metrics-mappings]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding component template [logs-settings]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [events]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics]
[00:10:04]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [logs]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-endpoint.metadata]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-endpoint.telemetry]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-endpoint.policy]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [events-endpoint]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.core]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [logs-system.auth]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.cpu]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.filesystem]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.fsstat]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.load]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.network_summary]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.diskio]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.network]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.memory]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.process_summary]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.entropy]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [logs-system.syslog]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.socket_summary]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.process]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.service]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.raid]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.users]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.uptime]
[00:10:05]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-debian-tests-xl-1591719320154896701] adding index template [metrics-system.socket]
[00:10:10]             └-: test metadata api
[00:10:10]               └-> "before all" hook
[00:10:10]               └-: POST /api/endpoint/metadata when index is not empty
[00:10:10]                 └-> "before all" hook
[00:10:10]                 └-> "before all" hook
[00:10:10]                   │ info [endpoint/metadata/api_feature] Loading "mappings.json"
[00:10:10]                   │ info [endpoint/metadata/api_feature] Loading "data.json"
[00:10:10]                   └- ✖ fail: "apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging""
[00:10:10]                   │

```

#### Stack Trace
```
{ Error: [illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition
    at respond (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:349:15)
    at checkRespForFailure (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:306:7)
    at HttpConnector.<anonymous> (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/connectors/http.js:173:7)
    at IncomingMessage.wrapper (/dev/shm/workspace/kibana/node_modules/elasticsearch/node_modules/lodash/lodash.js:4929:19)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  status: 400,
  displayName: 'BadRequest',
  message:
   '[illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition',
  path: '/metrics-endpoint.metadata-default-1',
  query: {},
  body:
   { error:
      { root_cause: [Array],
        type: 'illegal_argument_exception',
        reason:
         'mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition' },
     status: 400 },
  statusCode: 400,
  response:
   '{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"}],"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"},"status":400}',
  toString: [Function],
  toJSON: [Function] }
```
</details>

---

### History
* :green_heart: [Build #52637](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/) succeeded 106d0f28fd7ea147c61605eac732bba2ede133e4
* :green_heart: [Build #52525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/) succeeded bd1ca55211d03b14d95d0de2a349c98acce54f87
* :green_heart: [Build #52453](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/) succeeded 08b94cfe9f947f40dbdf43afe053b47d630153ea
* :green_heart: [Build #52304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/) succeeded acc3db620f9cb24e0db59d67c21e85bc9b4ca87a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/","number":"52304","commit":"acc3db620f9cb24e0db59d67c21e85bc9b4ca87a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/","number":"52453","commit":"08b94cfe9f947f40dbdf43afe053b47d630153ea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/","number":"52525","commit":"bd1ca55211d03b14d95d0de2a349c98acce54f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/","number":"52637","commit":"106d0f28fd7ea147c61605eac732bba2ede133e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/","number":"52756","commit":"143c0a8f550b776b0fc22e8f24009fcf4281c707"}]}
PIPELINE-->

ogupte commented on x-pack/plugins/apm/public/components/app/ServiceMap/cytoscapeOptions.ts:
```diff
@@ -183,7 +188,7 @@ const style: cytoscape.Stylesheet[] = [
   {
     selector: 'edge[isInverseEdge]',
-    style: { visibility: 'hidden' },
+    style: { visibility: 'none' },
```
i didn't see the comment, and i was getting the type error so i updated it...will change it back

ogupte committed:
- moved AnomalyDetection out of service map Contents into own component

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/)
* Commit: 7330a52dd11032910264e511f87ca299d466af25
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/execution/node/543/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/execution/node/698/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/testReport)

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/endpoint/metadata·ts.apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging"</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_endpoint_metadata·ts/Kibana_Pipeline___kibana_xpack_agent___apis_Endpoint_plugin_test_metadata_api_POST__api_endpoint_metadata_when_index_is_not_empty__before_all__hook_for__metadata_api_should_return_one_entry_for_each_host_with_default_paging_)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:09:40]           └-: Endpoint plugin
[00:09:40]             └-> "before all" hook
[00:09:40]             └-> "before all" hook
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [events-mappings]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [events-settings]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [logs-mappings]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [logs-settings]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [metrics-mappings]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [metrics-settings]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [logs]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [events]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-endpoint.policy]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-endpoint.metadata]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-endpoint.telemetry]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [events-endpoint]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [logs-system.auth]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.cpu]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.core]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.entropy]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.filesystem]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.diskio]
[00:09:41]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.fsstat]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.load]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.network_summary]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.network]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.process_summary]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.memory]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.raid]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [logs-system.syslog]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.uptime]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.process]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.service]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.socket]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.socket_summary]
[00:09:42]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.users]
[00:09:47]             └-: test metadata api
[00:09:47]               └-> "before all" hook
[00:09:47]               └-: POST /api/endpoint/metadata when index is not empty
[00:09:47]                 └-> "before all" hook
[00:09:47]                 └-> "before all" hook
[00:09:47]                   │ info [endpoint/metadata/api_feature] Loading "mappings.json"
[00:09:47]                   │ info [endpoint/metadata/api_feature] Loading "data.json"
[00:09:47]                   └- ✖ fail: "apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging""
[00:09:47]                   │

```

#### Stack Trace
```
{ Error: [illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition
    at respond (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:349:15)
    at checkRespForFailure (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:306:7)
    at HttpConnector.<anonymous> (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/connectors/http.js:173:7)
    at IncomingMessage.wrapper (/dev/shm/workspace/kibana/node_modules/elasticsearch/node_modules/lodash/lodash.js:4929:19)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  status: 400,
  displayName: 'BadRequest',
  message:
   '[illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition',
  path: '/metrics-endpoint.metadata-default-1',
  query: {},
  body:
   { error:
      { root_cause: [Array],
        type: 'illegal_argument_exception',
        reason:
         'mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition' },
     status: 400 },
  statusCode: 400,
  response:
   '{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"}],"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"},"status":400}',
  toString: [Function],
  toJSON: [Function] }
```
</details>

---

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/endpoint/metadata·ts.apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging"</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_endpoint_metadata·ts/Kibana_Pipeline___kibana_xpack_agent___apis_Endpoint_plugin_test_metadata_api_POST__api_endpoint_metadata_when_index_is_not_empty__before_all__hook_for__metadata_api_should_return_one_entry_for_each_host_with_default_paging__2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:10:20]           └-: Endpoint plugin
[00:10:20]             └-> "before all" hook
[00:10:20]             └-> "before all" hook
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [events-settings]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [events-mappings]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [logs-settings]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [logs-mappings]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [metrics-mappings]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding component template [metrics-settings]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [events]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [logs]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-endpoint.metadata]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-endpoint.policy]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-endpoint.telemetry]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [events-endpoint]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [logs-system.auth]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.core]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.cpu]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.diskio]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.fsstat]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.entropy]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.filesystem]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.memory]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.load]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.network_summary]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.network]
[00:10:21]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.process_summary]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.socket]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.service]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.raid]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.process]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [logs-system.syslog]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.socket_summary]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.uptime]
[00:10:22]               │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591731564154670381] adding index template [metrics-system.users]
[00:10:27]             └-: test metadata api
[00:10:27]               └-> "before all" hook
[00:10:28]               └-: POST /api/endpoint/metadata when index is not empty
[00:10:28]                 └-> "before all" hook
[00:10:28]                 └-> "before all" hook
[00:10:28]                   │ info [endpoint/metadata/api_feature] Loading "mappings.json"
[00:10:28]                   │ info [endpoint/metadata/api_feature] Loading "data.json"
[00:10:28]                   └- ✖ fail: "apis Endpoint plugin test metadata api POST /api/endpoint/metadata when index is not empty "before all" hook for "metadata api should return one entry for each host with default paging""
[00:10:28]                   │

```

#### Stack Trace
```
{ Error: [illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition
    at respond (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:349:15)
    at checkRespForFailure (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/transport.js:306:7)
    at HttpConnector.<anonymous> (/dev/shm/workspace/kibana/node_modules/elasticsearch/src/lib/connectors/http.js:173:7)
    at IncomingMessage.wrapper (/dev/shm/workspace/kibana/node_modules/elasticsearch/node_modules/lodash/lodash.js:4929:19)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  status: 400,
  displayName: 'BadRequest',
  message:
   '[illegal_argument_exception] mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition',
  path: '/metrics-endpoint.metadata-default-1',
  query: {},
  body:
   { error:
      { root_cause: [Array],
        type: 'illegal_argument_exception',
        reason:
         'mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition' },
     status: 400 },
  statusCode: 400,
  response:
   '{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"}],"type":"illegal_argument_exception","reason":"mapping fields [agent, endpoint, @timestamp, ecs, elastic, host, event] cannot be replaced during template composition"},"status":400}',
  toString: [Function],
  toJSON: [Function] }
```
</details>

---

### History
* :broken_heart: [Build #52756](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/) failed 143c0a8f550b776b0fc22e8f24009fcf4281c707
* :green_heart: [Build #52637](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/) succeeded 106d0f28fd7ea147c61605eac732bba2ede133e4
* :green_heart: [Build #52525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/) succeeded bd1ca55211d03b14d95d0de2a349c98acce54f87
* :green_heart: [Build #52453](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/) succeeded 08b94cfe9f947f40dbdf43afe053b47d630153ea
* :green_heart: [Build #52304](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/) succeeded acc3db620f9cb24e0db59d67c21e85bc9b4ca87a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52304/","number":"52304","commit":"acc3db620f9cb24e0db59d67c21e85bc9b4ca87a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/","number":"52453","commit":"08b94cfe9f947f40dbdf43afe053b47d630153ea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/","number":"52525","commit":"bd1ca55211d03b14d95d0de2a349c98acce54f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/","number":"52637","commit":"106d0f28fd7ea147c61605eac732bba2ede133e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/","number":"52756","commit":"143c0a8f550b776b0fc22e8f24009fcf4281c707"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/","number":"52839","commit":"7330a52dd11032910264e511f87ca299d466af25"}]}
PIPELINE-->

ogupte commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into apm-68485-fix-missing-service-ml-health-indicator

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52875/)
* Commit: aac21401550c7f5d9e804aa265d24f683b5eb6ed

### History
* :broken_heart: [Build #52839](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/) failed 7330a52dd11032910264e511f87ca299d466af25
* :broken_heart: [Build #52756](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/) failed 143c0a8f550b776b0fc22e8f24009fcf4281c707
* :green_heart: [Build #52637](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/) succeeded 106d0f28fd7ea147c61605eac732bba2ede133e4
* :green_heart: [Build #52525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/) succeeded bd1ca55211d03b14d95d0de2a349c98acce54f87
* :green_heart: [Build #52453](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/) succeeded 08b94cfe9f947f40dbdf43afe053b47d630153ea

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52453/","number":"52453","commit":"08b94cfe9f947f40dbdf43afe053b47d630153ea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/","number":"52525","commit":"bd1ca55211d03b14d95d0de2a349c98acce54f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/","number":"52637","commit":"106d0f28fd7ea147c61605eac732bba2ede133e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/","number":"52756","commit":"143c0a8f550b776b0fc22e8f24009fcf4281c707"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/","number":"52839","commit":"7330a52dd11032910264e511f87ca299d466af25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52875/","number":"52875","commit":"aac21401550c7f5d9e804aa265d24f683b5eb6ed"}]}
PIPELINE-->

ogupte commented:
retest

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52916/)
* Commit: aac21401550c7f5d9e804aa265d24f683b5eb6ed

### History
* :green_heart: [Build #52875](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52875/) succeeded aac21401550c7f5d9e804aa265d24f683b5eb6ed
* :broken_heart: [Build #52839](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/) failed 7330a52dd11032910264e511f87ca299d466af25
* :broken_heart: [Build #52756](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/) failed 143c0a8f550b776b0fc22e8f24009fcf4281c707
* :green_heart: [Build #52637](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/) succeeded 106d0f28fd7ea147c61605eac732bba2ede133e4
* :green_heart: [Build #52525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/) succeeded bd1ca55211d03b14d95d0de2a349c98acce54f87

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52525/","number":"52525","commit":"bd1ca55211d03b14d95d0de2a349c98acce54f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52637/","number":"52637","commit":"106d0f28fd7ea147c61605eac732bba2ede133e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52756/","number":"52756","commit":"143c0a8f550b776b0fc22e8f24009fcf4281c707"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52839/","number":"52839","commit":"7330a52dd11032910264e511f87ca299d466af25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52875/","number":"52875","commit":"aac21401550c7f5d9e804aa265d24f683b5eb6ed"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52916/","number":"52916","commit":"aac21401550c7f5d9e804aa265d24f683b5eb6ed"}]}
PIPELINE-->
